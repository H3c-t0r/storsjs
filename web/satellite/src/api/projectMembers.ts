// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

import apollo from '@/utils/apolloManager';
import gql from 'graphql-tag';

// Performs graqhQL request.
// Throws an exception if error occurs
export async function addProjectMember(userID: string, projectID: string): Promise<any> {
	let response: any = null;
	try {
		response = await apollo.mutate(
			{
				mutation: gql(`
                mutation {
                    addProjectMember(
                        projectID: "${projectID}",
                        userID: "${userID}"
                    ) {id}
                }`
				),
				fetchPolicy: 'no-cache',
			}
		);
	} catch (e) {
		// TODO: replace with popup in future
		console.error(e);
	}

	return response;
}

// Performs graqhQL request.
// Throws an exception if error occurs
export async function deleteProjectMember(userID: string, projectID: string): Promise<any> {
	let response: any = null;
	try {
		response = await apollo.mutate(
			{
				mutation: gql(`
                mutation {
                    deleteProjectMember(
                        projectID: "${projectID}",
                        userID: "${userID}"
                    ) {id}
                }`
				),
				fetchPolicy: 'no-cache',
			}
		);
	} catch (e) {
		// TODO: replace with popup in future
		console.error(e);
	}

	return response;
}

// Performs graqhQL request.
// Throws an exception if error occurs
export async function fetchProjectMembers(projectID: string, limit: string, offset: string): Promise<any> {
	let response: any = null;
	try {
		response = await apollo.query(
			{
				query: gql(`
                query {
                    project(
                        id: "${projectID}",
                    ) {
                        members(limit: ${limit}, offset: ${offset}) {
                            user {
                                id,
                                firstName,
                                lastName,
                                email,
                            },
                            joinedAt
                        }
                    }
                }`
				),
				fetchPolicy: 'no-cache',
			}
		);
	} catch (e) {
		// TODO: replace with popup in future
		console.error(e);
	}

	return response;
}
