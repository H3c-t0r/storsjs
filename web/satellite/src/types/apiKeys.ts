// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all apiKey-related functionality
 */
export interface ApiKeysApi {
    /**
     * Fetch apiKey
     *
     * @returns ApiKey
     * @throws Error
     */
    get(projectId: string): Promise<ApiKey[]>;
    create(projectId: string, name: string): Promise<ApiKey>;
    delete(ids: string[]): Promise<null>;
}

/**
 * ApiKey class holds info for ApiKey entity.
 */
export class ApiKey {
    public id: string;
    public secret: string;
    public name: string;
    public createdAt: string;
    public isSelected: boolean = false;

    constructor(id: string, name: string, createdAt: string, secret: string) {
        this.id = id || '';
        this.name = name || '';
        this.createdAt = createdAt || '';
        this.secret = secret || '';

        this.isSelected = false;
    }

    public formattedName(): string {
        let name = this.name;

        if (name.length < 12) {
            return name;
        }

        return name.slice(0, 12) + '...';
    }

    public getDate(): string {
        if (!this.createdAt) {
            return '';
        }

        return new Date(this.createdAt).toLocaleDateString();
    }
}
