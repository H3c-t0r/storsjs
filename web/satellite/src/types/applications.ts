// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

import VeeamIcon from '@/assets/apps/veeam.png';
import TrueNASIcon from '@/assets/apps/truenas.png';
import SplunkIcon from '@/assets/apps/splunk.png';
import ArqIcon from '@/assets/apps/arq.png';
import ZertoIcon from '@/assets/apps/zerto.png';
import UnitrendsIcon from '@/assets/apps/unitrends.png';
import S3FSIcon from '@/assets/apps/s3fs.png';
import RucioIcon from '@/assets/apps/rucio.png';
import PhotosPlusIcon from '@/assets/apps/photos+.png';
import MSP360Icon from '@/assets/apps/msp360.png';
import LucidLinkIcon from '@/assets/apps/lucidlink.png';
import IconikIcon from '@/assets/apps/iconik.png';
import HammerspaceIcon from '@/assets/apps/hammerspace.png';
import ElementsIcon from '@/assets/apps/elements.png';
import DockerIcon from '@/assets/apps/docker.png';
import PixelfedIcon from '@/assets/apps/pixelfed.png';
import MastodonIcon from '@/assets/apps/mastodon.png';
import DataverseIcon from '@/assets/apps/dataverse.png';
import CyberDuckIcon from '@/assets/apps/cyberduck.png';
import AtempoIcon from '@/assets/apps/atempo.png';
import AcronisIcon from '@/assets/apps/acronis.png';
import MountainDuckIcon from '@/assets/apps/mountainduck.png';
import FileZillaIcon from '@/assets/apps/filezilla.svg';
import GBLabsIcon from '@/assets/apps/gblabs.png';
import AdSignalIcon from '@/assets/apps/adsignal.png';
import OwnCloudIcon from '@/assets/apps/owncloud.png';
import LivepeerIcon from '@/assets/apps/livepeer.png';
import BunnyCDNIcon from '@/assets/apps/bunnycdn.png';
import CometIcon from '@/assets/apps/comet.png';
import FastlyIcon from '@/assets/apps/fastly.jpg';
import GlobusIcon from '@/assets/apps/globus.png';
import HuggingFaceIcon from '@/assets/apps/huggingface.svg';
import KerberosIcon from '@/assets/apps/kerberos.png';

export enum AppCategory {
    All = 'All',
    FileManagement = 'File Management',
    BackupRecovery = 'Backup & Recovery',
    ContentDistribution = 'Content Distribution',
    CDN = 'CDN',
    AI = 'AI',
    Logs = 'Logs',
    SocialMedia = 'Social Media',
}

export type Application = {
    title: string
    description: string
    category: AppCategory
    src: string
    docs: string
}

export const applications: Application[] = [
    {
        title: 'TrueNAS - iX Systems',
        description: 'Back up a single instance or keep several TrueNAS deployments synchronized using a single source of truth.',
        category: AppCategory.BackupRecovery,
        src: TrueNASIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ix-systems-truenas',
    },
    {
        title: 'Nebula by GB Labs',
        description: 'Cloud-based media storage solution powered by Storj. Work directly from the GB Labs Cloud with project files that are always in sync.',
        category: AppCategory.FileManagement,
        src: GBLabsIcon,
        docs: 'https://www.storj.io/partner-solutions/gb-labs',
    },
    {
        title: 'Hammerspace',
        description: 'Hammerspace and Storj bring more flexibility and agility to your global data landscape.',
        category: AppCategory.FileManagement,
        src: HammerspaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hammerspace',
    },
    {
        title: 'Ad Signal Match',
        description: 'Deduplicate assets and save on AI costs with Ad Signal Match. Transfer only your unique content to Storj.',
        category: AppCategory.FileManagement,
        src: AdSignalIcon,
        docs: 'https://www.storj.io/partner-solutions/ad-signal',
    },
    {
        title: 'OwnCloud',
        description: 'oCIS, or ownCloud Infinite Scale, is a cutting-edge technology platform for building cloud-native file sync and share applications.',
        category: AppCategory.FileManagement,
        src: OwnCloudIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/ocis',
    },
    {
        title: 'Livepeer',
        description: 'Add live and on-demand video experiences by transcoding videos with Livepeer and storing your media with Storj.',
        category: AppCategory.BackupRecovery,
        src: LivepeerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/livepeer',
    },
    {
        title: 'Acronis',
        description: 'Reliable backup and disaster recovery solutions for data archiving and organization, seamlessly integrating with Storj.',
        category: AppCategory.BackupRecovery,
        src: AcronisIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/acronis',
    },
    {
        title: 'Iconik',
        description: 'Iconik is easy to use and intuitive with a clean and intuitive platform that organizes your media and makes it searchable.',
        category: AppCategory.FileManagement,
        src: IconikIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/iconik',
    },
    {
        title: 'Photos+',
        description: 'Store and manage your photos and videos in your own cloud storage account.',
        category: AppCategory.FileManagement,
        src: PhotosPlusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/photos-plus',
    },
    {
        title: 'Veeam',
        description: 'All-in-one backup, recovery, and data security solution that serves both on-premises and cloud storage.',
        category: AppCategory.BackupRecovery,
        src: VeeamIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/veeam',
    },
    {
        title: 'Splunk',
        description: 'Splunk is a data analytics platform that provides data-driven insights across all aspects of a company.',
        category: AppCategory.Logs,
        src: SplunkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/splunk',
    },
    {
        title: 'Arq Backup',
        description: 'Arq backs up your filesystem with perfect, point-in-time backups of your files.',
        category: AppCategory.BackupRecovery,
        src: ArqIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/arq',
    },
    {
        title: 'Zerto',
        description: 'Easily incorporate immutability into a complete disaster recovery strategy.',
        category: AppCategory.BackupRecovery,
        src: ZertoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/zerto',
    },
    {
        title: 'Unitrends',
        description: 'Automate manual tasks, eliminate management complexity, and deliver tested hardware and software resilience.',
        category: AppCategory.BackupRecovery,
        src: UnitrendsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/unitrends',
    },
    {
        title: 'S3FS',
        description: 's3fs allows Linux, macOS, and FreeBSD to mount an S3 bucket via FUSE.',
        category: AppCategory.FileManagement,
        src: S3FSIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/s3fs',
    },
    {
        title: 'Rucio',
        description: 'Organize, manage, and access your data at scale with Rucio, an open-source framework for scientific collaboration.',
        category: AppCategory.FileManagement,
        src: RucioIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/rucio',
    },
    {
        title: 'MSP360',
        description: 'MSP360 is a best-in-class IT management platform for MSPs and internal IT teams.',
        category: AppCategory.BackupRecovery,
        src: MSP360Icon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/msp360',
    },
    {
        title: 'LucidLink',
        description: 'Using Storj with LucidLink provides resilient cloud object storage with blazing performance and zero-trust security.',
        category: AppCategory.FileManagement,
        src: LucidLinkIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/lucidlink',
    },
    {
        title: 'Elements',
        description: 'Store, manage and collaborate on media content using a secure and scalable asset management platform.',
        category: AppCategory.FileManagement,
        src: ElementsIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/elements',
    },
    {
        title: 'Docker',
        description: 'Storj supports custom Content-Type for any key, so it can be used as a container registry to distribute container images.',
        category: AppCategory.ContentDistribution,
        src: DockerIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/docker',
    },
    {
        title: 'Pixelfed',
        description: 'Learn how to set up Pixelfed decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: PixelfedIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/pixelfed',
    },
    {
        title: 'Mastodon',
        description: 'Here is how you can set up Mastodon decentralized social media platform to Storj.',
        category: AppCategory.SocialMedia,
        src: MastodonIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/mastodon',
    },
    {
        title: 'Dataverse',
        description: 'S3 credentials allow Dataverse to upload and download files from Storj as if it was using the S3 API.',
        category: AppCategory.FileManagement,
        src: DataverseIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/dataverse',
    },
    {
        title: 'CyberDuck',
        description: 'Open-source cloud storage browser app for macOS, Windows, and Linux that supports Storj.',
        category: AppCategory.FileManagement,
        src: CyberDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Atempo',
        description: 'Integrate with Storj to reduce storage costs without sacrificing security or performance for media archival with global access.',
        category: AppCategory.BackupRecovery,
        src: AtempoIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/atempo-miria',
    },
    {
        title: 'Mountain Duck',
        description: 'Add Storj as a disk on your computer. Open your Storj files with any application and work like on a local disk.',
        category: AppCategory.FileManagement,
        src: MountainDuckIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/cyberduck',
    },
    {
        title: 'Bunny CDN',
        description: 'Set up a static website hosting with Storj using Bunny CDN as a content delivery network providing a caching layer.',
        category: AppCategory.CDN,
        src: BunnyCDNIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/bunny',
    },
    {
        title: 'Fastly',
        description: 'Distribute your content among the Fastly edge cloud service using your Storj buckets as a source of content.',
        category: AppCategory.CDN,
        src: FastlyIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/fastly',
    },
    {
        title: 'Comet',
        description: 'Flexible backup platform integrates with Storj as a backup and storage destination to protect and restore data.',
        category: AppCategory.BackupRecovery,
        src: CometIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/comet',
    },
    {
        title: 'Globus',
        description: 'Seamless collaboration and management for data transfer, sharing, and discovery, all into one unified open-source platform.',
        category: AppCategory.FileManagement,
        src: GlobusIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/globus',
    },
    {
        title: 'HuggingFace',
        description: 'Train and deploy open-source AI models, while saving and loading datasets to and from Storj.',
        category: AppCategory.AI,
        src: HuggingFaceIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/hugging-face',
    },
    {
        title: 'FileZilla',
        description: 'Learn how to set up FileZilla to transfer files over Storj DCS or integrate FileZilla Pro.',
        category: AppCategory.FileManagement,
        src: FileZillaIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/filezilla',
    },
    {
        title: 'Kerberos.io',
        description: 'Open-source platform for video analytics and management. Integrate with Storj to store your Kerberos Vault video files.',
        category: AppCategory.FileManagement,
        src: KerberosIcon,
        docs: 'https://docs.storj.io/dcs/third-party-tools/kerberos-vault',
    },
];
