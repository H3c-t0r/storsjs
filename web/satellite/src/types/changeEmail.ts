// Copyright (C) 2024 Storj Labs, Inc.
// See LICENSE for copying information.

export enum ChangeEmailStep {
    InitStep = 0,
    VerifyPasswordStep,
    Verify2faStep,
    VerifyOldEmailStep,
    SetNewEmailStep,
    VerifyNewEmailStep,
    SuccessStep,
}
