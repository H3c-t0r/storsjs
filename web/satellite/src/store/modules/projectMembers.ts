// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

import { PROJECT_MEMBER_MUTATIONS } from '../mutationConstants';
import { addProjectMember, deleteProjectMember, fetchProjectMembers } from '@/api/projectMembers';

export const projectMembersModule = {
	state: {
		projectMembers: [],
	},
	mutations: {
		[PROJECT_MEMBER_MUTATIONS.ADD](state: any, projectMember: TeamMemberModel) {
			state.projectMembers.push(projectMember);
		},
		[PROJECT_MEMBER_MUTATIONS.DELETE](state: any, projectMemberId: string) {

			const projectMember = state.projectMembers
				.find((teamMember: TeamMemberModel) => teamMember.user.id === projectMemberId);

			const index = state.projectMembers.indexOf(projectMember, 0);
			if (index === -1) {
				// TODO: Notify about error

				return;
			}

			state.projectMembers.splice(index, 1);
		},
		[PROJECT_MEMBER_MUTATIONS.TOGGLE_SELECTION](state: any, projectMemberId: string) {
			state.projectMembers = state.projectMembers.map((projectMember: any) => {
				if (projectMember.user.id === projectMemberId) {
					projectMember.isSelected = !projectMember.isSelected;
				}

				return projectMember;
			});
		},
		[PROJECT_MEMBER_MUTATIONS.CLEAR_SELECTION](state: any) {
			state.projectMembers = state.projectMembers.map((projectMember: any) => {
				projectMember.isSelected = false;

				return projectMember;
			});
		},
		[PROJECT_MEMBER_MUTATIONS.FETCH](state: any, teamMembers: any[]) {
			state.projectMembers = teamMembers;
		},
	},
	actions: {
		addProjectMember: async function ({commit, rootState}: any, userId: string): Promise<boolean> {
			const projectId = rootState.projectsModule.state.selectedProject.id;
			const response = await addProjectMember(userId, projectId);

			if (!response || !response.data) {
				return false;
			}

			commit(PROJECT_MEMBER_MUTATIONS.ADD, response.data);

			return true;
		},
		deleteProjectMembers: async function ({commit, rootGetters}: any, projectMemberIds: string[]): Promise<boolean> {
			const projectId = rootGetters.selectedProject.id;
			let isDeletionSuccess = true;

			for await (const id of projectMemberIds) {
				const response = await deleteProjectMember(id, projectId);

				if (!response || !response.data) {
					isDeletionSuccess = false;
				}

				commit(PROJECT_MEMBER_MUTATIONS.DELETE, id);
			}

			return isDeletionSuccess;
		},
		toggleProjectMemberSelection: function ({commit}: any, projectMemberId: string) {
			commit(PROJECT_MEMBER_MUTATIONS.TOGGLE_SELECTION, projectMemberId);
		},
		clearProjectMemberSelection: function ({commit}: any) {
			commit(PROJECT_MEMBER_MUTATIONS.CLEAR_SELECTION);
		},
		fetchProjectMembers: async function ({commit, rootGetters}: any): Promise<boolean> {
			const projectId = rootGetters.selectedProject.id;
			const response = await fetchProjectMembers(projectId);

			if (!response || !response.data) {
				return false;
			}

			commit(PROJECT_MEMBER_MUTATIONS.FETCH, response.data.project.members);

			return true;
		},
	},
	getters: {
		projectMembers: (state: any) => state.projectMembers,
		selectedProjectMembers: (state: any) => state.projectMembers.filter((member: any) => member.isSelected)
	},
};
