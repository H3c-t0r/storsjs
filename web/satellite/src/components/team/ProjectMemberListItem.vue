// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="user-container">
        <div class="user-container__base-info">
            <div class="checkbox"></div>
            <div class="user-container__base-info__avatar" :style="avatarData.style">
                <h1 class="user-container__base-info__avatar__letter">{{avatarData.letter}}</h1>
            </div>
            <div class="user-container__base-info__name-area">
                <p class="user-container__base-info__name-area__user-name">{{itemName}}</p>
                <p v-if="isProjectOwner" class="user-container__base-info__name-area__owner-status">Project Owner</p>
            </div>
        </div>
        <p class="user-container__date">{{itemDate}}</p>
        <p class="user-container__user-email">{{itemEmail}}</p>
    </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

import { ProjectMember } from '@/types/projectMembers';
import { getColor } from '@/utils/avatarColorManager';

@Component
export default class ProjectMemberListItem extends Vue {
    @Prop({default: new ProjectMember('', '', '', '', '')})
    public itemData: ProjectMember;

    public get avatarData(): object {
        const fullName: string = this.itemData.user.getFullName();

        const letter = fullName.slice(0, 1).toLocaleUpperCase();

        const style = {
            background: getColor(letter),
        };

        return {
            letter,
            style,
        };
    }

    public get itemName(): string {
        return this.itemData.formattedFullName();
    }

    public get itemDate(): string {
        return this.itemData.joinedAtLocal();
    }

    public get itemEmail(): string {
        return this.itemData.formattedEmail();
    }

    public get isProjectOwner(): boolean {
        return this.itemData.user.id === this.$store.getters.selectedProject.ownerId;
    }
}
</script>

<style scoped lang="scss">
    .user-container {
        display: flex;
        flex-direction: row;
        align-items: center;
        padding-left: 28px;
        height: 83px;
        background-color: #fff;
        cursor: pointer;
        width: calc(100% - 28px);
        font-family: 'font_regular';

        &__base-info {
            width: 50%;
            display: flex;
            align-items: center;
            justify-content: flex-start;

            &__avatar {
                min-width: 40px;
                max-width: 40px;
                min-height: 40px;
                max-height: 40px;
                margin-left: 20px;
                border-radius: 6px;
                display: flex;
                align-items: center;
                justify-content: center;
                background-color: #FF8658;

                &__letter {
                    margin: 0;
                    font-size: 16px;
                    color: #F5F6FA;
                }
            }

            &__name-area {

                &__user-name {
                    margin: 0 0 0 20px;
                    font-size: 16px;
                    font-family: 'font_bold';
                    color: #354049;
                }

                &__owner-status {
                    margin: 0 0 0 20px;
                    font-size: 13px;
                    color: #AFB7C1;
                    font-family: 'font_medium';
                }
            }
        }

        &__date {
            width: 25%;
            font-size: 16px;
            color: #354049;
        }

        &__user-email {
            width: 25%;
            font-size: 16px;
            color: #354049;
        }
    }

    .checkbox {
        background-image: url("../../../static/images/team/checkboxEmpty.svg");
        min-width: 23px;
        height: 23px;
    }

    .user-container.selected {
        background-color: #2683FF;

        .checkbox {
            min-width: 23px;
            height: 23px;
            background-image: url("../../../static/images/team/checkboxChecked.svg");
        }

        .user-container__base-info__name-area__user-name,
        .user-container__base-info__name-area__owner-status,
        .user-container__date,
        .user-container__user-email {
            color: #FFFFFF;
        }
    }
</style>
