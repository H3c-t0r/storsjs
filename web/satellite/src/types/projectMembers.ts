// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

// ProjectMember stores needed info about user info to show it on UI
import { User } from '@/types/users';
import { ProjectMemberSortByEnum, ProjectMemberSortDirectionEnum } from '@/utils/constants/ProjectMemberSortEnum';

export type OnHeaderClickCallback = (sortBy: ProjectMemberSortByEnum, sortDirection: ProjectMemberSortDirectionEnum) => Promise<any>;

export class ProjectMemberCursor {
    public search: string;
    public limit: number;
    public page: number;
    public order: ProjectMemberSortByEnum;
    public orderDirection: ProjectMemberSortDirectionEnum;

    public constructor() {
        this.search = '';
        this.limit = 6;
        this.page = 1;
        this.order = ProjectMemberSortByEnum.NAME;
        this.orderDirection = ProjectMemberSortDirectionEnum.ASCENDING;
    }
}

export class ProjectMembersPage {
    public projectMembers: ProjectMember[];
    public search: string;
    public order: ProjectMemberSortByEnum;
    public orderDirection: ProjectMemberSortDirectionEnum;
    public limit: number;
    public pageCount: number;
    public currentPage: number;
    public totalCount: number;

    public constructor() {
        this.projectMembers = [];
        this.search = '';
        this.order = ProjectMemberSortByEnum.NAME;
        this.orderDirection = ProjectMemberSortDirectionEnum.ASCENDING;
        this.limit = 8;
        this.pageCount = 0;
        this.currentPage = 1;
        this.totalCount = 0;
    }
}

export class ProjectMember {
    public user: User;

    public joinedAt: string;
    public isSelected: boolean;

    public constructor(fullName: string, shortName: string, email: string, joinedAt: string, id?: string) {
        this.user = new User(fullName, shortName, email);
        this.user.id = id || '';
        this.joinedAt = joinedAt;
    }

    public formattedFullName(): string {
        let fullName: string = this.user.getFullName();

        if (fullName.length > 16) {
            fullName = fullName.slice(0, 13) + '...';
        }

        return fullName;
    }

    public formattedEmail(): string {
        let email: string = this.user.email;

        if (email.length > 16) {
            email = this.user.email.slice(0, 13) + '...';
        }

        return email;
    }

    public joinedAtLocal(): string {
        if (!this.joinedAt) return '';

        return new Date(this.joinedAt).toLocaleDateString();
    }
}

