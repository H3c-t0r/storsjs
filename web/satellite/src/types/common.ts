// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { Validator } from '@/utils/validation';
import { useConfigStore } from '@/store/modules/configStore';

export enum SortDirection {
    asc = 1,
    desc = 2,
}

export class PricingPlanInfo {
    constructor(
        public type: PricingPlanType = PricingPlanType.FREE,
        // Info for the pricing plan container
        public title: string = '',
        public containerSubtitle: string = '',
        public containerDescription: string = '',
        public containerFooterHTML: string | null = null,
        public activationButtonText: string | null = null,
        // Info for the pricing plan modal (pre-activation)
        public activationSubtitle: string | null = null,
        public activationDescriptionHTML: string = '',
        public activationPriceHTML: string | null = null,
        // Info for the pricing plan modal (post-activation)
        public successSubtitle: string = '',
        public bannerTitle: string = '',
        public bannerText: string = '',
    ) {}
}

export interface OnboardingInfo {
    accessText?: string;
    accessBtnText?: string;
    accessTitle?: string;
}

export enum PricingPlanType {
    FREE = 'free',
    PARTNER = 'partner',
    PRO = 'pro',
}

// TODO: fully implement these types and their methods according to their Go counterparts
export type UUID = string
export type MemorySize = string
export type Time = string

export function tableSizeOptions(itemCount: number, isObjectBrowser = false): {title: string, value: number}[] {
    const opts = [
        { title: '10', value: 10 },
        { title: '25', value: 25 },
        { title: '50', value: 50 },
        { title: '100', value: 100 },
    ];
    if (itemCount <= 300 && !isObjectBrowser) {
        return [{ title: 'All', value: itemCount }, ...opts];
    }
    return opts;
}

export type ValidationRule<T> = string | boolean | ((value: T) => string | boolean);

export function RequiredRule(value: unknown): string | boolean {
    return (Array.isArray(value) ? !!value.length : !!value) || 'Required';
}

export function EmailRule(value: string, strict = false): string | boolean {
    return Validator.email(value, strict) || 'E-mail must be valid.';
}

export function MaxNameLengthRule(value: string): string | boolean {
    const { maxNameCharacters } = useConfigStore().state.config;

    return Validator.nameLength(value, maxNameCharacters) || `The value must be less than or equal to ${maxNameCharacters}.`;
}

export interface IDialogFlowStep {
    onEnter?: () => void;
    onExit?: (to: 'next' | 'prev') => void;
    validate?: () => boolean;
}

export interface DialogStepComponent extends IDialogFlowStep {
    title: string;
    iconSrc?: string;
}

export type SaveButtonsItem = string | {
    name: string;
    value: string;
};

export const MAX_SEARCH_VALUE_LENGTH = 200;

export function getUniqueName(name: string, allNames: string[]): string {
    // Regular expression to match a name with an optional numeric suffix.
    const namePattern = /^(.*?)(?: \((\d+)\))?$/;

    let currName = name;
    let count = 0;
    while (allNames.includes(currName)) {
        count++;
        currName = currName.replace(namePattern, (_, baseName, index) => {
            // Increment the suffix if it exists, or add a new one starting with 1.
            const newIndex = index ? parseInt(index) + 1 : count;
            return `${baseName} (${newIndex})`;
        });
    }

    return currName;
}
