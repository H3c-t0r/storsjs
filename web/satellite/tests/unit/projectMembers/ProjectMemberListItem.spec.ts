// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import ProjectMemberListItem from '@/components/team/ProjectMemberListItem.vue';

import { ProjectMember } from '@/types/projectMembers';
import { mount } from '@vue/test-utils';

describe('', () => {
    it('should renders correctly', function () {
        const member: ProjectMember = new ProjectMember('testFullName', 'testShortName', 'test@example.com', '2019-08-09T08:52:43.695679Z', '1');

        const wrapper = mount(ProjectMemberListItem, {
            propsData: {
                itemData: member,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render correctly with item row highlighted', function () {
        const member: ProjectMember = new ProjectMember('testFullName', 'testShortName', 'test@example.com', '2019-08-09T08:52:43.695679Z', '1');
        member.isSelected = true;

        const wrapper = mount(ProjectMemberListItem, {
            propsData: {
                itemData: member,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
