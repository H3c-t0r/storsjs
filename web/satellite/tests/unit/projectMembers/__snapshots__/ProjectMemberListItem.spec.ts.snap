// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with item row highlighted 1`] = `
<div class="user-container">
  <div class="user-container__base-info">
    <div class="checkbox"></div>
    <div class="user-container__base-info__avatar" style="background: rgb(88, 185, 255);">
      <h1 class="user-container__base-info__avatar__letter">T</h1>
    </div>
    <p class="user-container__base-info__user-name">testShortName</p>
  </div>
  <p class="user-container__date">8/9/2019</p>
  <p class="user-container__user-email">test@example.com</p>
</div>
`;

exports[`should renders correctly 1`] = `
<div class="user-container">
  <div class="user-container__base-info">
    <div class="checkbox"></div>
    <div class="user-container__base-info__avatar" style="background: rgb(88, 185, 255);">
      <h1 class="user-container__base-info__avatar__letter">T</h1>
    </div>
    <p class="user-container__base-info__user-name">testShortName</p>
  </div>
  <p class="user-container__date">8/9/2019</p>
  <p class="user-container__user-email">test@example.com</p>
</div>
`;
